Source/WebCore/page/Quirks.cpp

```diff
@@ -593,10 +580,7 @@ bool Quirks::shouldDispatchedSimulatedMouseEventsAssumeDefaultPrevented(EventTar
     if (!needsQuirks() || !shouldDispatchSimulatedMouseEvents(target))
         return false;
 
-    if (!m_quirksData.shouldDispatchedSimulatedMouseEventsAssumeDefaultPreventedQuirk)
-        m_quirksData.shouldDispatchedSimulatedMouseEventsAssumeDefaultPreventedQuirk = isAmazon() || isSoundCloud();
-
-    if (!m_quirksData.shouldDispatchedSimulatedMouseEventsAssumeDefaultPreventedQuirk.value())
+    if (!isAmazon() || !isSoundCloud())
```

Comment:
Perhaps we could just remove this early return? (Since we'll fall through to the final `return false` anyways)

Reply:
Well, actually, this avoids a potential downcast. So I suspect we want to keep this as-is.